packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -1359,6 +1359,10 @@ export default class InternalModel {
 
     if (didChange && id !== null) {
       this.store.setRecordId(this.modelName, id, this.clientId);
+      // internal set of ID to get it to RecordData from DS.Model
+      if (this._recordData.__setId) {
+        this._recordData.__setId(id);
```

Comment:
may want to leave yourself a note to use optional chaining here when TS 3.7 is safe to adopt

Reply:
Thanks, want to leave the check more loud for the reader/explicit in this case